### PR TITLE
Update pipeline to independently build A6 or A7

### PR DESCRIPTION
**Pull request by KSerrania**
### What does this PR do?

- Update container tagging strategy: provide hints to extract the correct git tag for the right agent (ie. A6 version should be 6.x.y, while A7 will be 7.x.y). The invoke scripts already support this with the `PYTHON_RUNTIMES` environment variable.   
- Skip A6 or A7 jobs if not building a package for a given version. This should apply to triggered builds only, otherwise all relevant jobs should run. When triggered, we will use two new environment variables `RELEASE_VERSION_6` and `RELEASE_VERSION_7`.
  - This is based off a new `except` gitlab YAML clause that should be typically paired with the `run_if_triggered` "family" conditional sections within the YAML job descriptions. 

### Motivation

Agent 6 & Agent 7 concurrent & independent builds.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -161,15 +163,15 @@ before_script:
     - echo "Running windows source tests"
   tags: ["runner:windows-docker", "windowsversion:1809"]
   script:
-    - docker run --rm -m 4096M -v "$(Get-Location):c:\mnt" -e IS_AWS_CONTAINER=true -e SIGN_WINDOWS=true 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-builders/windows_${ARCH}:$Env:DATADOG_AGENT_WINBUILDIMAGES \unittests.bat --release-version %RELEASE_VERSION% 
+    - docker run --rm -m 4096M -v "$(Get-Location):c:\mnt" -e IS_AWS_CONTAINER=true -e SIGN_WINDOWS=true 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-builders/windows_${ARCH}:$Env:DATADOG_AGENT_WINBUILDIMAGES \unittests.bat
```
Comment (albertvaka) [inline .gitlab-ci.yml]: Is this change intentional? Why?

**Review by albertvaka — APPROVED** [review]

File: .gitlab-ci.yml (hunk 1 above)
Comment (KSerrania) [reply .gitlab-ci.yml]: Yes, the `unittests.bat` script doesn't use this option
